tests/test_mpi.py

```diff
@@ -1005,6 +1008,183 @@ def test_avoid_haloupdate_if_distr_but_sequential(self, mode):
         calls = FindNodes(Call).visit(op)
         assert len(calls) == 0
 
+    @pytest.fixture
+    def setup(self):
+        shape = (2,)
+        so = 2
+        tn = 30
+
+        grid = Grid(shape=shape)
+
+        # Velocity and pressure fields
+        v = TimeFunction(name='v', grid=grid, space_order=so)
+        tau = TimeFunction(name='tau', grid=grid, space_order=so)
+
+        # First order elastic-like dependencies equations
+        pde_v = v.dt - (tau.dx)
+        pde_tau = tau.dt - ((v.forward).dx)
+        u_v = Eq(v.forward, solve(pde_v, v.forward))
+        u_tau = Eq(tau.forward, solve(pde_tau, tau.forward))
+
+        # Receiver
+        rec = SparseTimeFunction(name="rec", grid=grid, npoint=1, nt=tn)
+        rec.coordinates.data[:, 0] = np.linspace(0., shape[0], num=1)
+
+        return grid, v, tau, u_v, u_tau, rec
+
+    @pytest.mark.parallel(mode=1)
+    def test_issue_2448_I(self, mode, setup):
+        _, v, tau, u_v, u_tau, rec = setup
+
+        rec_term0 = rec.interpolate(expr=v)
+
+        op0 = Operator([u_v, u_tau, rec_term0])
+
+        calls = [i for i in FindNodes(Call).visit(op0) if isinstance(i, HaloUpdateCall)]
+
+        assert len(calls) == 3
+        assert len(FindNodes(HaloUpdateCall).visit(op0.body.body[1].body[1].body[0])) == 1
+        assert len(FindNodes(HaloUpdateCall).visit(op0.body.body[1].body[1].body[1])) == 2
+        assert calls[0].arguments[0] is v
+        assert calls[1].arguments[0] is tau
+        assert calls[2].arguments[0] is v
+
+    @pytest.mark.parallel(mode=1)
+    def test_issue_2448_II(self, mode, setup):
```

Comment:
leaving the comment here but it applies to this entire new code segment. SInce it's a large batch of tests all around issue 2448, you could think of creating a class `class TestIssue2448` 

Then the name `setup` for the fixture would become acceptable

And you could add a docstring for more general explanation

Reply:
Another possibility:

I just saw that there's a new `TestElastic`...

maybe call it `TestElasticLike` and move this whole batch of new tests in there?

Up to you